src/gui/databasekey/PasswordEditWidget.cpp

```diff
@@ -103,16 +105,6 @@ void PasswordEditWidget::initComponent()
            "<p>Good passwords are long and unique. KeePassXC can generate one for you.</p>"));
 }
 
-void PasswordEditWidget::hideEvent(QHideEvent* event)
-{
-    if (!isVisible() && m_compUi->enterPasswordEdit) {
-        m_compUi->enterPasswordEdit->setText("");
-        m_compUi->repeatPasswordEdit->setText("");
-    }
-
-    QWidget::hideEvent(event);
-}
```

Comment:
This was added on purpose in cb2900f5a98b3774b9dfaa4c6ead4f1a3b18c6fc and it also triggers on events such as window minimisation (at least tray minimisation). I'm unsure whether we should actually remove this. We should at least fix the password clear timer.

Reply:
In my testing, this caused data loss when switching between the tabs on the security page. I decided to compromise by doing the clearing only when the user explicitly cancels the action to change the password. Remember, this isn't the actual password "yet".

Reply:
I made some improvements to the clearing process when you "cancel" the database settings.